frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-internal/status-card/status-card.tsx

```diff
@@ -74,6 +80,75 @@ export const StatusCard: React.FC<DashboardItemProps> = ({
     t,
   );
 
+  const [isHCPresent, setHCPresence] = React.useState(false);
+  const healthChecksGetterOrPresent = (isPresent: boolean = false) => {
+    const cephDetails = data?.[0]?.status?.ceph?.details;
+    const pattern = /[A-Z]+_*|error/g;
+    const healthChecks: CephHealthCheck[] = [];
+    for (const key in cephDetails) {
+      if (pattern.test(key)) {
+        if (isPresent) {
+          return true;
+        }
+        const healthCheckObject: CephHealthCheck = {
+          details: cephDetails[key].message,
+          troubleshootLink: whitelistedHealthChecksRef[key] ?? null,
+        };
+        healthChecks.push(healthCheckObject);
+      }
+    }
+    return isPresent ? false : healthChecks;
+  };
+
+  React.useEffect(
+    () => {
+      setHCPresence(healthChecksGetterOrPresent(true) as boolean);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [data?.[0]?.status?.ceph?.details],
+  );
+
+  const CephHealthChecks: React.FC = () => {
+    const healthChecks = healthChecksGetterOrPresent();
```

Comment:
Its not good to define components inside components.
Move this outside. Pass ceph data in `props`